### PR TITLE
#1362 & #1364 fix: refresh in user profile & admin page userlist

### PR DESCRIPTION
**Pull request by reyhanrumengan**

### CODE REVIEW

File: hms-app/src/components/HeaderMenu.tsx [hunk 1]
```diff
@@ -65,8 +66,19 @@ export default function HeaderMenu({
...
+    lastName: 'string',
+    firstName: 'string',
+    emailAddress: 'string',
+    roles: [],
+    skills: [],
+    imageUrl: 'string',
+    creationDate: new Date(),
+  } as User)
+  const [isUserError, setIsUserError] = useState(false)
+  const [isUserLoading, setIsUserLoading] = useState(true)
```
Comment (LetsPoke) [inline hms-app/src/components/HeaderMenu.tsx:81]: both states are never used

File: hms-app/src/components/HeaderMenu.tsx [hunk 2]
```diff
@@ -76,6 +88,24 @@ export default function HeaderMenu({
...
+      () => {
+        setIsUserLoading(false)
+        setIsUserError(true)
+      }
+    )
+  }
+
+  useEffect(() => {
+    loadSelectedUser()
+  }, [userDetails.id])
```
Comment (LetsPoke) [inline hms-app/src/components/HeaderMenu.tsx:107]: this looks weird, when does the userDetails.id change?

File: hms-app/src/components/card-details/UserDetails.tsx [hunk 3]
```diff
@@ -40,7 +36,6 @@ export default function UserDetails(props: { userId: string }) {
   } as User)
 
   const loadSelectedUser = () => {
-    setIsUserLoading(true)
```
Comment (LetsPoke) [inline hms-app/src/components/card-details/UserDetails.tsx:43]: why is this deleted?

File: hms-app/src/components/input-forms/EditUserForm.tsx [hunk 4]
```diff
@@ -11,12 +11,15 @@ import { Check, X } from 'tabler-icons-react'
...
+  reload?: () => void
+  setOpened?: (boolean: boolean) => void
 }
 
 export default function EditUserForm(props: IProps) {
   const { instance } = useMsal()
   const { classes } = styles()
-  const { userId } = props
+  const { userId, reload, setOpened } = props
+  // const { userId } = props
```
Comment (LetsPoke) [inline hms-app/src/components/input-forms/EditUserForm.tsx:22]: pls remove this comment

File: hms-app/src/components/HeaderMenu.tsx (hunk 1 above)
Comment (reyhanrumengan) [reply hms-app/src/components/HeaderMenu.tsx:81]: do we need it?

File: hms-app/src/components/card-details/UserDetails.tsx (hunk 3 above)
Comment (reyhanrumengan) [reply hms-app/src/components/card-details/UserDetails.tsx:43]: after everytime we edit, there was a little bit of glitch. 
Only after I delete this part, its not happening again

**Comment by reyhanrumengan** [comment]
`export const deleteUser = async (
  instance: IPublicClientApplication,
  userID: string
) => {
  const idToken = await getIdToken(instance)
  const options = buildFetchOptions('DELETE', idToken)
  return fetch(`${coreUrl}/hackathon/${userID}`, options)
    .then((response) => {
      return response.json()
    })
    .catch((err) => console.log(err))
}`

Does this code in the UserAction correct? Why is it "/hackathon/${userID}"? Is it not "/user/${userID}"
In the console, it also says "DELETE localhost:3000/local/hackathon/...... 404 (Not Found)"

@LetsPoke

File: hms-app/src/components/HeaderMenu.tsx (hunk 1 above)
Comment (LetsPoke) [reply hms-app/src/components/HeaderMenu.tsx:81]: maybe we should wrap the return statement with this error and loading checks? so that if something goes wrong we can display a "whoopsie" message. 
take a look at UserDetails line 111

File: hms-app/src/components/card-details/UserDetails.tsx (hunk 3 above)
Comment (LetsPoke) [reply hms-app/src/components/card-details/UserDetails.tsx:43]: could the glitch be the loading time of the user? 
so can we say "while user is loading display loading...."?

**Comment by LetsPoke** [comment]
> `export const deleteUser = async ( instance: IPublicClientApplication, userID: string ) => { const idToken = await getIdToken(instance) const options = buildFetchOptions('DELETE', idToken) return fetch(`${coreUrl}/hackathon/${userID}`, options) .then((response) => { return response.json() }) .catch((err) => console.log(err)) }`
> 
> Does this code in the UserAction correct? Why is it "/hackathon/${userID}"? Is it not "/user/${userID}" In the console, it also says "DELETE localhost:3000/local/hackathon/...... 404 (Not Found)"
> 
> @LetsPoke

yeah that one is wrong :O 
it should be user/{id}

File: hms-app/src/components/card-details/UserDetails.tsx (hunk 3 above)
Comment (reyhanrumengan) [reply hms-app/src/components/card-details/UserDetails.tsx:43]: We already have the that in the return statement. 
I just checked and try to see it in slow motion, and turns out it does shows "User details are loading". But because it only happen for like a few millisecond, we saw it as a glitch.

File: hms-app/src/components/HeaderMenu.tsx [hunk 5]
```diff
@@ -204,77 +239,103 @@ export default function HeaderMenu({
...
-          </Button>  
-        </Popover.Target>
-        <Popover.Dropdown>
-          <Text size='sm'>
-            Name: {userDetails?.firstName} {userDetails?.lastName}
-          </Text>
+        </Button>
+      </Popover.Target>
+      <Popover.Dropdown>
+        {isUserError && !isUserLoading && (
```
Comment (LetsPoke) [inline hms-app/src/components/HeaderMenu.tsx:268]: maybe easier to read if isUserLoading is always the first check

File: hms-app/src/components/HeaderMenu.tsx [hunk 6]
```diff
@@ -65,8 +66,19 @@ export default function HeaderMenu({
   today.setHours(0, 0, 0, 0)
   const location = useLocation()
   const { instance, accounts } = useMsal()
```
Comment (LetsPoke) [inline hms-app/src/components/HeaderMenu.tsx:68]: accounts here is never used

File: hms-app/src/components/HeaderMenu.tsx [hunk 7]
```diff
@@ -76,6 +88,25 @@ export default function HeaderMenu({
...
+      () => {
+        setIsUserLoading(false)
+        setIsUserError(true)
+      }
+    )
+  }
+
+  useEffect(() => {
+    loadSelectedUser()
+  }, [])
```
Comment (LetsPoke) [inline hms-app/src/components/HeaderMenu.tsx:108]: shouldnt this one be called when EditModalOpened changes?

**Review by LetsPoke — APPROVED** [review]
